docs/esa/hubble.rst

```diff
@@ -239,11 +260,55 @@ If no criteria are specified to limit the selection, this function will retrieve
 
 This will perform a cone search with radius 7 arcmins. The result of the
 query will be returned and stored in the votable file
-'cone_search_m31_5.vot'.
+'cone_search_m31_5.vot'. If no filename is defined and the "save" tag is True,
+the module will provide a default name. It is also possible to store only the results
+in memory, without defining neither a filename nor the "save" tag.
+
+----------------------------------------------------
+7. Cone searches with criteria in the Hubble archive
+----------------------------------------------------
+
+It is also possible to perform a cone search defined by a target name or coordinates, a radius
+and a set of criteria to filter the results.
+
+.. code-block:: python
+
+  >>> from astroquery.esa.hubble import ESAHubble
+  >>> esahubble = ESAHubble()
+  >>> result = esahubble.cone_search_criteria(target= 'm31',radius=7,
+                                              obs_collection=['HST'],
+                                              data_product_type = 'image',
+                                              instrument_name = ['ACS/WFC'],
+                                              filters = ['F435W'],
+                                              async_job = True,
+                                              filename = 'output1.vot.gz',
+                                              output_format="votable")
+  >>> str(table)
+
+.. code-block:: python
+
+  >>> from astropy import coordinates
+  >>> from astroquery.esa.hubble import ESAHubble
+  >>> esahubble = ESAHubble()
+  >>> c = coordinates.SkyCoord("00h42m44.51s +41d16m08.45s", frame='icrs')
+  >>> result = esahubble.cone_search_criteria(coordinates=c,radius=7,
+                                              obs_collection=['HST'],
+                                              instrument_name = ['WFPC2'],
+                                              filters = ['F606W'],
+                                              async_job = True,
+                                              filename = 'output1.vot.gz',
+                                              output_format="votable",
+                                              verbose=True)
+  >>> str(table)
+
+This will perform a cone search with radius 7 arcmins around the target (defined by
+its coordinates or its name) using the filters defined when executing the function.
+
+This function allows the same parameters than the search criteria (see Section 5).
 
 
 -------------------------------
-7. Getting access to catalogues
+8. Getting access to catalogues
```

Comment:
you may consider using rst/sphinx to auto numerate these headings, but it's also fine to leave as is.

Reply:
I will leave it as is for this PR, maybe I will remove it in the following iteration (it seems that enumerations are not widely used in other modules...).